vmm/src/cpu.rs

```diff
@@ -1171,11 +1307,24 @@ struct CPU {
     cpu_id: u8,
 }
 
-#[cfg(feature = "acpi")]
+#[cfg(all(target_arch = "x86_64", feature = "acpi"))]
 const MADT_CPU_ENABLE_FLAG: usize = 0;
 
 #[cfg(feature = "acpi")]
 impl Aml for CPU {
+    #[cfg(target_arch = "aarch64")]
+    fn to_aml_bytes(&self) -> Vec<u8> {
+        aml::Device::new(
+            format!("C{:03}", self.cpu_id).as_str().into(),
+            vec![
+                &aml::Name::new("_HID".into(), &"ACPI0007"),
+                &aml::Name::new("_UID".into(), &self.cpu_id),
+            ],
+        )
+        .to_aml_bytes()
+    }
```

Comment:
Ultimately I think you'll want the methods to support hotplug (or can you not hotplug CPUs on aarcht64?) so maybe you could reuse more of the version below but just exclude the _MAT field.

Reply:
Agree. Now I am reusing the code as much as possible. There should be some problems, errors were seen in guest kernel messages. More work is needed. But no test cases were broken. So I think we can live with these errors now, soon or later we will handle them.